pkg/gofr/datasource/mongo/mongo.go

```diff
@@ -101,28 +102,36 @@ func (c *Client) Connect() {
 
 // InsertOne inserts a single document into the specified collection.
 func (c *Client) InsertOne(ctx context.Context, collection string, document interface{}) (interface{}, error) {
-	defer c.sendOperationStats(&QueryLog{Query: "insertOne", Collection: collection, Filter: document}, time.Now())
+	tracedCtx, span := c.addTrace(ctx, "insert", collection)
 
-	return c.Database.Collection(collection).InsertOne(ctx, document)
+	result, err := c.Database.Collection(collection).InsertOne(tracedCtx, document)
+
+	defer c.pushObservability(&QueryLog{Query: "insertOne", Collection: collection, Filter: document}, time.Now(),
+		"insert", span)
```

Comment:
not necessary, but IMO we should not move `defer pushObservability` from beginning to end..
if updating, please update everywhere..

Reply:
I have updated it everywhere.